test/question.spec.js

```diff
@@ -0,0 +1,196 @@
+      expect(resp_data.message).to.equal('Successfully edited question');
+      expect(resp_data.question).to.be.an('object');
+    });
+    it('Should search question', async () => {
+      const search_query = 'question';
+      const response = await getRequest(`/questions/search/${search_query}?page=0&limit=10`, token).expect(200);
+
+      const resp_data = response.body;
+      expect(resp_data).to.be.an('object');
+      expect(resp_data).to.have.property('message');
+      expect(resp_data).to.have.property('questions');
+      expect(resp_data.message).to.be.an('string');
+      expect(resp_data.message).to.equal('Successfully got questions.');
+      expect(resp_data.questions).to.be.an('array');
+    });
+    it('Should get question', async () => {
+      const question = await fetchQuestion({ id: question_id }, true);
+
+      expect(question).to.be.an('object');
+      expect(question).to.have.property('question_text');
+    });
+    it('Should get questions', async () => {
+      const questions = await fetchQuestions(true);
+
+      expect(questions).to.be.an('array');
+    });
+    it('Should get questions with most answers', async () => {
+      const questions = await fetchQuestionsWithMostAnswers(question_id);
+      expect(questions).to.be.an('array');
+    });
+    it('Should delete a question', async () => {
+      const response = await deleteRequest(`/questions/${question_id}`, token).expect(200);
+
+      const resp_data = response.body;
+      expect(resp_data).to.be.an('object');
+      expect(resp_data).to.have.property('message');
+      expect(resp_data.message).to.be.an('string');
+      expect(resp_data.message).to.equal('Successfully deleted question');
+    });
+    after(async () => {
+      const userToDelete = await fetchUser({ username: user.username });
+      await destroyUser(userToDelete.id, true);
+    });
+  });
+
+  describe('Negative Tests', () => {
+    let token;
+    const user = {
+      username: 'kufre',
+      firstname: 'Kufre',
+      lastname: 'Okon',
+      email: `${Date.now()}_example@example.com`,
+      password: 'test123',
+    };
+    before(async () => {
+      const password = await hashPassword('test123');
+      await createUser({ ...user, password });
+      const response = await request(app)
+        .post('/auth/login')
+        .send(user)
+        .set('Accept', 'application/json');
+      token = response.body.accessToken;
+    });
+    it('Should not post a new question with empty question_text', async () => {
+      const response = await postRequest('/questions', token)
+        .send({
+          question_text: '',
+        })
+        .expect(500);
+
+      const resp_data = response.body;
+      expect(resp_data).to.be.an('object');
+      expect(resp_data).to.have.property('error');
+      expect(resp_data.error).to.be.an('string');
+      expect(resp_data.error).to.equal('notNull Violation: questions.question_text cannot be null');
```

Comment:
This error message is not really nice to the end user, you would need to setup even if a manual check in the controller like `if(!req.body.question_text) return res.status(400).json({message:'Question text is required'})`.

** You can skip for now if time won't permit